src/express-server/mongomodels/product.ts

```diff
@@ -0,0 +1,25 @@
+import { model, Schema } from "mongoose";
+
+const ProductSchema: Schema = new Schema({
+  id: {
+    type: String,
+    required: true,
+  },
+  title: {
+    type: String,
+    required: true,
+    minlength: [5, "Too short product title"],
+  },
+  price: String,
+  reviews: {
+    type: String,
+    default: "No reviews for the product",
+  },
+  lastModifiedDate: Date,
+});
+
+ProductSchema.pre("save", function () {
+  this["lastModifiedDate"] = new Date();
```

Comment:
The issue is than `pre('save')` isn't triggered for `update` events, so it will only work for creation

Reply:
As far as I remember, there is no ability to update products. But for cities pre('findOneAndUpdate') was added.